### PR TITLE
Co2 model

### PR DESCRIPTION
**Pull request by cruizm**

### CODE REVIEW

File: test/co2_model_lab/CO2_coupled.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+using state=logger::logger<logger::logger_state, dynamic::logger::formatter<TIME>, oss_sink_state>;
+using log_messages=logger::logger<logger::logger_messages, dynamic::logger::formatter<TIME>, oss_sink_messages>;
+using global_time_mes=logger::logger<logger::logger_global_time, dynamic::logger::formatter<TIME>, oss_sink_messages>;
+using global_time_sta=logger::logger<logger::logger_global_time, dynamic::logger::formatter<TIME>, oss_sink_state>;
+
+using logger_top=logger::multilogger<state, log_messages, global_time_mes, global_time_sta>;
+
+int main() {
+
+    grid_scenario<co2, int> scenario = grid_scenario<co2,int>({38, 57}, co2(), false);
```
Comment (romancardenas) [inline test/co2_model_lab/CO2_coupled.cpp:54]: The same: You don't have to define the vicinity type

File: test/co2_model_lab/CO2_coupled.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                break;             
+            case -200:
+                input.type = CO2_SOURCE;
+                break;
+            default:
+                assert(false && "invalid input");
+            }
+                
+ 
+        scenario.set_initial_state({cell_x, cell_y}, co2(input.counter, input.concentration, input.type));
```
Comment (romancardenas) [inline test/co2_model_lab/CO2_coupled.cpp:116]: This configuration is nice. We should think about an alternative way using JSON format

File: test/hoya/hoya_coupled.cpp [hunk 3]
```diff
@@ -58,6 +58,6 @@ int main() {
     std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> t = std::make_shared<grid_coupled<TIME, sir, mc>>(test);
```
Comment (romancardenas) [inline test/hoya/hoya_coupled.cpp:58]: The file is named `hoya_coupled.cpp`. You should rename it to match the CO2 model

File: test/co2_model_lab/CO2_coupled.cpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    }
+};
+
+using state=logger::logger<logger::logger_state, dynamic::logger::formatter<TIME>, oss_sink_state>;
+using log_messages=logger::logger<logger::logger_messages, dynamic::logger::formatter<TIME>, oss_sink_messages>;
+using global_time_mes=logger::logger<logger::logger_global_time, dynamic::logger::formatter<TIME>, oss_sink_messages>;
+using global_time_sta=logger::logger<logger::logger_global_time, dynamic::logger::formatter<TIME>, oss_sink_state>;
+
+using logger_top=logger::multilogger<state, log_messages, global_time_mes, global_time_sta>;
+
```
[the remark this answers is not in the document]
Comment (cruizm) [reply test/co2_model_lab/CO2_coupled.cpp:51]: Totally agree

File: test/co2_model_lab/CO2_coupled.cpp (hunk 2 above)
Comment (cruizm) [reply test/co2_model_lab/CO2_coupled.cpp:116]: Totally agree. This was just a quick fix to be able to parse the original val file

**Pull request merged**